docs/sharing-configurations.md

```diff
@@ -0,0 +1,183 @@
+
+Before we start, make sure you have:
+
+- An account for [npmjs.com](https://www.npmjs.com/) to publish the package
+- Basic understating about [how to create a Node.js module](https://docs.npmjs.com/creating-node-js-modules)
+
+First, create a new package. We recommend creating a [scoped package](https://docs.npmjs.com/cli/v10/using-npm/scope) with the name `@username/prettier-config`.
+
+A minimal package should have at least two files. A `package.json` for the package configuration and an `index.js` which holds the shared prettier configuration object:
+
+```text
+prettier-config/
+├── index.js
+└── package.json
+```
+
+Example `package.json`:
+
+```json
+{
+  "name": "@username/prettier-config",
+  "version": "1.0.0",
+  "description": "My personal Prettier config",
+  "type": "module",
+  "exports": "./index.js",
+  "license": "MIT",
+  "peerDependencies": {
+    "prettier": ">=2.0.0"
+  }
+}
+```
+
+`index.js` is where you put the shared configuration. This file just exports a [regular prettier configuration](./configuration.md) with the same syntax and same options:
+
+```js
+const config = {
+  trailingComma: "es5",
+  tabWidth: 4,
+  singleQuote: true,
+};
+
+export default config;
+```
+
+An example shared configuration repository is available [here](https://github.com/azz/prettier-config).
+
+## Publishing a Shareable Config
+
+Once you are ready, you can [publish your package to npm](https://docs.npmjs.com/creating-and-publishing-scoped-public-packages#publishing-scoped-public-packages):
+
+```sh
+npm publish --access public
+```
+
+## Using a Shareable Config
+
+You first need to install your published configuration, for example:
+
+```sh
+npm install @username/prettier-config --save-dev
+```
+
+Then, you can reference it in your `package.json`:
+
+```json
+{
+  "name": "my-cool-library",
+  "version": "9000.0.1",
+  "prettier": "@username/prettier-config"
+}
+```
+
+If you don’t want to use `package.json`, you can use any of the supported extensions to export a string, e.g. `.prettierrc.json`:
+
+```json
+"@company/prettier-config"
+```
+
+### Extending a Sharable Config
+
+To _extend_ the configuration to overwrite some properties from the shared configuration, import the file in a `.prettierrc.mjs` file and export the modifications, e.g:
+
+```js
+import myPrettierConfig from "@username/prettier-config";
+
+export default {
+  ...myPrettierConfig,
+  semi: false,
+};
+```
+
+## Other examples
+
+### Using Type Annotation in the Shared Config
+
+You can get type safety and autocomplete support in your shared configuration by using a `jsdoc` type annotation:
+
+```js
+/**
+ * @type {import("prettier").Config}
+ */
+const config = {
+  trailingComma: "es5",
+  tabWidth: 4,
+  semi: false,
+  singleQuote: true,
+};
+
+export default config;
+```
+
+In order to make this work, you have to install `prettier` as a `devDependency` for the project:
+
+```sh
+npm install -D prettier
+```
```

Comment:
Let's link to install page https://prettier.io/docs/en/install

Reply:
https://github.com/prettier/prettier/pull/16495/commits/72230349ed989b6b2c50d702c5081affaf95770e ✅